sdks/python/apache_beam/runners/dataflow/internal/dependency.py

```diff
@@ -488,6 +488,23 @@ def _stage_beam_sdk_tarball(sdk_remote_location, staged_path, temp_dir):
         'type of location: %s' % sdk_remote_location)
 
 
+def get_runner_harness_container_image():
+  """For internal use only; no backwards-compatibility guarantees.
+
+   Returns:
+     str: Runner harness container image that shall be used by default
+       for current SDK version or None if the runner harness container image
+       bundled with the service shall be used.
+  """
+  import pkg_resources as pkg
+  try:
+    version = pkg.get_distribution(GOOGLE_PACKAGE_NAME).version
+    return (DATAFLOW_CONTAINER_IMAGE_REPOSITORY + '/' + 'harness' + ':' +
+            str(version))
```

Comment:
<!--thread_id:cc_143607416_t; commit:0df823c8839461643514dc954b59c2e073253fb0; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> Do we trust the behavior of the `str()` method across different versions of setuptools?  In particular, this way of stringifying the version is different from the method on line 542.

<!--section:body-->
I think we can trust this behavior. str actually is not needed here since version field is already a string.  I think we should use plain string representation version of the SDK version and assume that each Dataflow/Beam release has a correspondent version of containers, and simplify the logic.